app/controllers/orders_controller.rb

```diff
@@ -19,10 +19,18 @@ def create # rubocop:disable Metrics/MethodLength, Metrics/AbcSize
     @order = Order.new(permitted_attributes.merge(created_by: current_user))
     authorize @order
 
+    current_credit = @order.user.credit
+
     if @order.save
+      if (@order.user.provider == 'amber_oauth2') && @order.user.credit.negative? && current_credit.positive?
```

Comment:
Will current_credit.positive not just point to the same value that @order.user.credit.positive points to? Am I missing something here?
Does @order.save have side effects?

Reply:
I wondered this too, but it retrieves the credit from the database again. Not sure if it always does this or just after saving (didn't test), but it does work in this context.